### PR TITLE
Upgrade bindgen.

### PR DESCRIPTION
**Pull request by frewsxcv**
This was needed to get things compiling on Apple Silicon.

https://github.com/rust-lang/rust-bindgen/blob/master/CHANGELOG.md

### CODE REVIEW

File: src/proj.rs [hunk 1]
```diff
@@ -792,16 +793,17 @@ impl Proj {
...
                 proj_errno_reset(self.c_proj);
                 trans =
-                    proj_trans_array(self.c_proj, PJ_DIRECTION_PJ_FWD, pj.len(), pj.as_mut_ptr());
+                    proj_trans_array(self.c_proj, PJ_DIRECTION_PJ_FWD, pj_len, pj.as_mut_ptr());
                 err = proj_errno(self.c_proj);
             },
             Transformation::Projection => unsafe {
                 proj_errno_reset(self.c_proj);
-                trans = proj_trans_array(self.c_proj, inv, pj.len(), pj.as_mut_ptr());
+                trans = proj_trans_array(self.c_proj, inv, pj_len, pj.as_mut_ptr());
```
Comment (frewsxcv) [inline src/proj.rs:806]: https://github.com/rust-lang/rust-bindgen/pull/1688

**Comment by urschrei** [comment]
Bors try

File: src/proj.rs (hunk 1 above)
Comment (michaelkirk) [reply src/proj.rs:806]: Won't this break against our prebuilt bindings?

I admit - I'm kind of surprised by the changes made to `bindgen`.

Comment (michaelkirk) [reply src/proj.rs:806]: I suppose we can just re-generate all the prebuilt bindings with the new bindgen.

Comment (michaelkirk) [reply src/proj.rs:806]: 🤦 Oh yeah, I never did https://github.com/georust/proj/issues/44, so we don't have any prebuilt bindings. 😅

Comment (frewsxcv) [reply src/proj.rs:806]: > I admit - I'm kind of surprised by the changes made to `bindgen`.

there appears to be an ongoing discussions about that change https://github.com/rust-lang/rust-bindgen/issues/1901

**Comment by frewsxcv** [comment]
Yay tests pass

**Comment by michaelkirk** [comment]
> Yay tests pass on CI (still not locally)

Hrmm... could you paste the bindings.rs that are being generated and the error you're seeing? I'm super invested in the future of usize vs. bindgen vs M1 now 😅 

(reminded to self: that we'll probably want to update `bindings_docs-rs.rs` whenever we update bindgen in case the output changes other semantics. I think this is just a manual process of copying bindings.rs from the target/*/proj-sys*/ directory)

**Comment by urschrei** [comment]
> I think this is just a manual process of copying bindings.rs from the target//proj-sys/ directory)

Yep, it was a very sophisticated workaround…

**Comment by frewsxcv** [comment]
Just double checking, are there any outstanding questions here? I opted in to `.size_t_is_usize(true)`, so there's minimal changes in the pull request

**Comment by michaelkirk** [comment]
> Hrmm... could you paste the bindings.rs that are being generated and the error you're seeing? I'm super invested in the future of size_t vs. bindgen vs M1 now 😅

LGTM! 

I'm still curious as to what specific failure you were seeing on your M1 and would love to peep the bindings if you still somehow have them.

**Comment by frewsxcv** [comment]
bors r=michaelkirk

**Comment by frewsxcv** [comment]
> > Hrmm... could you paste the bindings.rs that are being generated and the error you're seeing? I'm super invested in the future of size_t vs. bindgen vs M1 now 😅
> 
> LGTM!
> 
> I'm still curious as to what specific failure you were seeing on your M1 and would love to peep the bindings if you still somehow have them.

I don't unfortunately :-/

**Comment by frewsxcv** [comment]
bors retry

**Comment by michaelkirk** [comment]
huh. I reran the action and CI passed. I fear for our flapping CI.

bors retry

**Pull request merged**